<!DOCTYPE html>
<html lang="en"> 
<head>
    <title>CReSCENT Documentation</title>
    
    <!-- Meta -->
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    <meta name="description" content="CReSCENT Docs - User Manual for scRNA-seq workflows on CReSCENT">
    <meta name="author" content="Xiaoying Riley at 3rd Wave Media">    
    <link rel="shortcut icon" href="favicon.ico"> 
    
    <!-- Google Font -->
    <link href="https://fonts.googleapis.com/css?family=Poppins:300,400,500,600,700&display=swap" rel="stylesheet">
    
    <!-- FontAwesome JS-->
    <script defer src="assets/fontawesome/js/all.min.js"></script>
    
    <!-- Plugins CSS -->
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.2/styles/atom-one-dark.min.css">

    <!-- Theme CSS -->  
    <link id="theme-style" rel="stylesheet" href="assets/css/theme.css">

</head> 

<body class="docs-page">    
    <header class="header fixed-top">	    
        <div class="branding docs-branding">
            <div class="container-fluid position-relative py-2">
                <div class="docs-logo-wrapper">
					<button id="docs-sidebar-toggler" class="docs-sidebar-toggler docs-sidebar-visible mr-2 d-xl-none" type="button">
	                    <span></span>
	                    <span></span>
	                    <span></span>
	                </button>
									<div class="site-logo"><a class="navbar-brand" href="index.html""><img class="logo-icon mr-2" src="assets/images/CReSCENT-svg.svg" style="height:40px;width:50px;" alt="logo"><span class="logo-text">CReSCENT<span class="text-alt">Docs</span></span></a></div>    
									
                </div><!--//docs-logo-wrapper-->
	            <div class="docs-top-utilities d-flex justify-content-end align-items-center">
	                <div class="top-search-box d-none d-lg-flex">
		                <form class="search-form">
				            <input type="text" placeholder="Search the docs..." name="search" class="form-control search-input">
				            <button type="submit" class="btn search-btn" value="Search"><i class="fas fa-search"></i></button>
				        </form>
	                </div>
	
					<ul class="social-list list-inline mx-md-3 mx-lg-5 mb-0 d-none d-lg-flex">
						<li class="list-inline-item"><a href="https://github.com/pughlab/crescent-frontend" target="_blank"><i class="fab fa-github fa-fw"></i></a></li>
		                <li class="list-inline-item"><a href="https://youtu.be/7a23FASMn6I" target="_blank"><i class="fab fa-youtube"></i></a></li>
		            </ul><!--//social-list-->
					<a href="https://crescent.cloud/" target="_blank" class="btn btn-primary"><i class="fas fa-play-circle mr-2"></i>Go to crescent.cloud!</a>
					<!-- <a href="#" class="btn btn-info"><i class="fas fa-play-circle mr-2"></i> Go to crescent.cloud</a> -->

	            </div><!--//docs-top-utilities-->
            </div><!--//container-->
        </div><!--//branding-->
    </header><!--//header-->
    
    <div class="docs-wrapper">
	    <div id="docs-sidebar" class="docs-sidebar">
		    <div class="top-search-box d-lg-none p-3">
                <form class="search-form">
		            <input type="text" placeholder="Search the docs..." name="search" class="form-control search-input">
		            <button type="submit" class="btn search-btn" value="Search"><i class="fas fa-search"></i></button>
		        </form>
            </div>
		    <nav id="docs-nav" class="docs-nav navbar">
			    <ul class="section-items list-unstyled nav flex-column pb-3">
				    <li class="nav-item section-title"><a class="nav-link scrollto active" href="#section-1"><span class="theme-icon-holder mr-2"><i class="fas fa-info"></i></span>Introduction</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-1-1">About us</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-1-2">Contact us</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-1-3">Portal overview</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-2"><span class="theme-icon-holder mr-2"><i class="fas fa-folder-open"></i></span>Creating a project</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-2-1">Uploading scRNA-seq data</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-2-2">Uploading cell metadata</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-3"><span class="theme-icon-holder mr-2"><i class="fas fa-paper-plane"></i></span>Creating a run</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-3-1">Configuring parameters</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-3-2">Run status</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-4"><span class="theme-icon-holder mr-2"><i class="fas fa-spinner"></i></span>Visualizing results</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-4-1">Quality control</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-4-2">Gene expression</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-5"><span class="theme-icon-holder mr-2"><i class="fas fa-arrow-down"></i></span>Downloading results</a></li>
				    <li class="nav-item"><a class="nav-link scrollto" href="#item-5-1">Failed run logs</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-6"><span class="theme-icon-holder mr-2"><i class="fas fa-user-plus"></i></span>Sharing a project</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-7"><span class="theme-icon-holder mr-2"><i class="fas fa-trash"></i></span>Deleting projects and runs</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-8"><span class="theme-icon-holder mr-2"><i class="fas fa-book-open"></i></span>Resources</a></li>
				    <li class="nav-item section-title mt-3"><a class="nav-link scrollto" href="#section-9"><span class="theme-icon-holder mr-2"><i class="fas fa-lightbulb"></i></span>FAQs</a></li>
			    </ul>

		    </nav><!--//docs-nav-->
	    </div><!--//docs-sidebar-->
	    <div class="docs-content">
		    <div class="container">
				<article class="docs-article" id="section-1">
				    <header class="docs-header">
						<!-- <h1 class="docs-heading">CReSCENT: <br/>CanceR Single Cell ExpressioN Toolkit</h1> -->
						<h1 class="docs-heading">Introduction <span class="docs-time">Last updated: 2020-03-25 16:37 EST</span></h1>
					</header>
					<section class="docs-section" id='item-1-1'>
						<h2 class="section-heading">About us</h2>
						<p><b>CReSCENT</b>: CanceR Single Cell ExpressioN Toolkit (<a href="https://crescent.cloud/" target="_blank">https://crescent.cloud</a>), is an intuitive and scalable web portal incorporating a containerized pipeline execution engine for standardized analysis of single-cell RNA sequencing (scRNA-seq) data. While scRNA-seq data for cancer specimens is easily generated, subsequent analysis requires high-performance computing infrastructure and user expertise to tailor interpretation for cancer. CReSCENT is populated with multiple public datasets and preconfigured pipelines that are accessible to computational biology non-experts, and user-editable to allow for optimization, comparison, and reanalysis on the fly.</p>
						<p>CReSCENT is supported and funded by Genome Canada, Medicine by Design, Princess Margaret Cancer Centre, the University Health Network, and the Hospital for Sick Children.</p>
					</section><!--//docs-intro-->
					<section class="docs-section" id="item-1-2">
						<h2 class="section-heading">Contact us</h2>
						<p>We'd like to hear from you! Please feel free to reach the CReSCENT team to report bugs, suggest features, etc. Email us at <a href="mailto:crescent@uhnresearch.ca">crescent@uhnresearch.ca</a> or submit an issue to our <a href="https://github.com/pughlab/crescent-frontend/issues" target="_blank"><i class="fab fa-github"></i>Github</a>.</p>
					</section><!--//section-->
				     <section class="docs-section" id="item-1-3">
						<h2 class="section-heading">Portal overview</h2>
						<img class="img-fluid" src="assets/images/intro_3_main_pages.jpg">
						<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>The three main pages within CReSCENT: 1) Projects/home, 2) Project runs, 3) Visualizations.</figcaption>
						<p><br/>The CReSCENT interface is represented by three main pages: projects, runs, and visualizations. The projects page is the homepage and here you can upload your own data or explore public cancer datasets curated by the CReSCENT team. Once a project is selected, the runs page shows all the runs/jobs that have been submitted. Run statuses are displayed by the run card colour (e.g. green means completed, yellow means in progress, etc.). The runs page also houses functional buttons for sharing the current project with other CReSCENT users as well as uploading metadata to the project. Once a run is selected, the visualizations page displays the available visualizations including quality control plots and explorable gene expression through t-SNE, UMAP, and violin plots. Actions within these three pages will be detailed throughout the documentation.</p>
					</section><!--//section-->
			    </article><!--//docs-article-->
				

			    <article class="docs-article" id="section-2">
				    <header class="docs-header">
					    <h1 class="docs-heading">Creating a project</h1>
					    <section class="docs-intro">
							<p>CReSCENT is organized into projects and each project features a single, scRNA-seq dataset. Projects exist within the portal as public or uploaded (private). Public projects are curated by the CReSCENT team and results within these projects can be explored. In addition to exploring these public projects, your own scRNA-seq data can be uploaded and processed. </p>
						</section><!--//docs-intro-->
				    </header>
				     <section class="docs-section" id="item-2-1">
						<h2 class="section-heading">Uploading scRNA-seq data</h2>
						<p>Single cell RNA-seq data is uploaded upon the creation of a new project. A raw or normalized count matrix can be easily uploaded to the portal in MTX format. File formats are further described below. </p>
						<!-- <h5>Upload example</h5> -->
						<figure class="figure docs-figure py-3">
							<a href="assets/images/create_project.gif" data-title="Uploading scRNA-seq data" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/create_project.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Quick walkthrough of how to upload your own scRNA-seq data and create a project in CReSCENT.</figcaption>
						</figure>					
						<h3>File format</h3>
						<p>CReSCENT currently accepts scRNA-seq data in the <a href="https://math.nist.gov/MatrixMarket/formats.html" target="_blank">Matrix Market (MTX) format</a> as input. Compared to gene-by-barcode text files, the MTX format requires less storage space for sparse matrices where many elements are zeros. The three files used as input are the MTX (matrix.mtx.gz) and its barcode and gene/feature indices (barcodes.tsv.gz and features.mtx.gz). These files are a typical output from a <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices" target="_blank">10X Cell Ranger</a> run.</p>
						<h5>Gene-by-Barcode Matrix:</h5>
						<div class="table-responsive my-4">
							<table class="table table-striped">
								<thead>
									<tr>
										<th scope="col"></th>
										<th scope="col">BARCODE1</th>
										<th scope="col">BARCODE2</th>
										<th scope="col">BARCODE3</th>
									</tr>
								</thead>
								<tbody>
									<tr>
										<th scope="row">GENE1</th>
										<td>0</td>
										<td>1</td>
										<td>0</td>
									</tr>
									<tr>
										<th scope="row">GENE2</th>
										<td>2</td>
										<td>1</td>
										<td>7</td>
									</tr>
									<tr>
										<th scope="row">GENE3</th>
										<td>1</td>
										<td>4</td>
										<td>0</td>
									</tr>
								</tbody>
							</table>
						</div><!--//table-responsive-->
						<div class="callout-block callout-block-info">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-info-circle"></i>
		                            </span><!--//icon-holder-->
	                                Note
	                            </h4>
								<p>One-line-command script to convert gene-by-barcode text files to MTX files can be found <a href="https://github.com/pughlab/crescent/blob/master/bin/development/obtains_mtx_files_from_gene_x_barcode_matrix.R" target="_blank">here</a>. Example data from 10X Genomics is also available! The scRNA-seq dataset has PBMCs with approximately 1000 cells. It can be downloaded from our portal when creating a project or click <a href="https://crescent.cloud/express/download/5e5818142f7290004cab4868" target="_blank">here</a> to download! </p>
							</div><!--//content-->
						</div><!--//callout-block-->
						<div class="callout-block callout-block-warning">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-bullhorn"></i>
		                            </span><!--//icon-holder-->
	                                Warning
	                            </h4>
								<p>Personal/protected health information (PHI) is prohibited from upload. Do not include patient names, dates of birth, MRNs and other patient identifiers in your uploads. Examples of PHI can be found <a href="https://privacyruleandresearch.nih.gov/pr_08.asp" target="_blank">here</a>. Only de-identified data is accepted in CReSCENT! </p>
							</div><!--//content-->
                        </div><!--//callout-block-->

					</section><!--//section-->
					
					<section class="docs-section" id="item-2-2">
						<h2 class="section-heading">Uploading cell metadata</h2>
	
						<p>Following project creation, cell metadata can be uploaded if available. Cell metadata will be used to visualize and colour cells as well as provide further insight into gene expression exploration in the visualizations page. </p>
						<!-- <h5>Upload example</h5> -->
						<figure class="figure docs-figure py-3">
							<a href="assets/images/upload_metadata.gif" data-title="Uploading cell metadata" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/upload_metadata.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Quick walkthrough of how to upload cell metadata for a project.</figcaption>
						</figure>
				
						<h3>File format</h3>
						<p>CReSCENT currently accepts cell metadata in a tab-separated file (tsv). We have adopted the <a href="https://github.com/broadinstitute/single_cell_portal/wiki/Metadata-File" target="_blank">Broad standard</a> for cell metadata. This tsv has cell barcodes in the first column and customized metadata labels in the subsequent columns. These labels can be categorized as "group" for categorical variables, like cell types, and "numeric" for continuous values, like orthogonal assay measurements. The double header is necessary to display the proper visualization with "NAME" and "TYPE" being mandatory. </p>
						<h5>Cell Metadata:</h5>
						<div class="table-responsive my-4">
							<table class="table table-striped">
								<thead>
									<tr>
										<th scope="col">NAME</th>
										<th scope="col">Cell_Type</th>
										<th scope="col">Cell_Phase</th>
										<th scope="col">Mitochondrial_Fraction</th>
									</tr>
									<tr>
										<th scope="col">TYPE</th>
										<th scope="col">group</th>
										<th scope="col">group</th>
										<th scope="col">numeric</th>
									</tr>
								</thead>
								<tbody>
									<tr>
										<th scope="row">BARCODE1</th>
										<td>Endothelial_cell</td>
										<td>G1</td>
										<td>0.174</td>
									</tr>
									<tr>
										<th scope="row">BARCODE2</th>
										<td>Fibroblast</td>
										<td>S</td>
										<td>0.054</td>
									</tr>
									<tr>
										<th scope="row">BARCODE3</th>
										<td>Cardiac_muscle_cell</td>
										<td>G2</td>
										<td>0.199</td>
									</tr>
								</tbody>
							</table>
						</div><!--//table-responsive-->
						<div class="callout-block callout-block-info">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-info-circle"></i>
		                            </span><!--//icon-holder-->
	                                Note
	                            </h4>
								<p>This is an optional upload! Some examples include orthogonal assays, sample annotations, cell annotations, etc.</p>
							</div><!--//content-->
						</div><!--//callout-block-->
						<div class="callout-block callout-block-warning">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-bullhorn"></i>
		                            </span><!--//icon-holder-->
	                                Warning
	                            </h4>
								<p>Personal/protected health information (PHI) is prohibited from upload. Do not include patient names, dates of birth, MRNs and other patient identifiers in your uploads. Examples of PHI can be found <a href="https://privacyruleandresearch.nih.gov/pr_08.asp" target="_blank">here</a>. Only de-identified data is accepted in CReSCENT! </p>
							</div><!--//content-->
                        </div><!--//callout-block-->
					</section><!--//section-->
			    </article><!--//docs-article-->
			    
			    
			    <article class="docs-article" id="section-3">
				    <header class="docs-header">
					    <h1 class="docs-heading">Creating a run</h1>
					    <section class="docs-intro">
							<p>Once you have created a CReSCENT project, you can now start submitting runs! A run involves optionally setting parameters for a standardized single cell analysis pipeline that includes quality control, normalization, dimension reduction, cell clustering, and differential gene expression. The pipeline is preconfigured but parameters are user-editable to allow for optimization, comparison, and reanalysis on the fly.</p> 
							<figure class="figure docs-figure py-3">
								<a href="assets/images/name_run.gif" data-title="Creating a run" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/name_run.gif" alt="" style="width: 800px;"></a>
								<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Creating and naming a project run in CReSCENT.</figcaption>
							</figure>	
						</section><!--//docs-intro-->
				    </header>
				     <section class="docs-section" id="item-3-1">
						<h2 class="section-heading">Configuring parameters</h2>
						<p>Once a project run is created, parameters can be configured, and the run can be submitted for analysis. Each step within the pipeline has its own parameters (i.e. quality control step has a mitochondrial content parameter). These parameters can be changed to adjust for your scRNA-seq sample (i.e. adjusting for higher mitochondrial content due to the nature of the tissue). Multiple runs can also be submitted to optimize and compare results from differing parameters (i.e. too high of a mitochondrial content can include dying or cells in stress and this can be tweaked in different runs to see its effect on results). Parameter details are described in the CReSCENT portal above every parameter.</p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/set_parameters.gif" data-title="Configuring parameters" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/set_parameters.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Quick walkthrough of how to configure pipeline parameters for a project run and submit the run. </figcaption>
						</figure>
						<div class="callout-block callout-block-info">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-info-circle"></i>
		                            </span><!--//icon-holder-->
	                                Note
	                            </h4>
								<p>Parameter details are described in the CReSCENT portal above every parameter.</p>
							</div><!--//content-->
						</div><!--//callout-block-->	
					</section><!--//section-->
					
					<section class="docs-section" id="item-3-2">
						<h2 class="section-heading">Run status</h2>
						<p>Run status is indicated by the card colour. In other words, an orange run card represents a run that has not been submitted, a yellow run card represents a run being processed, a green run card represents a completed run with results available for exploration, and finally, a red run card represents a run that has failed with logs available for download. </p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/run_status.gif" data-title="Run status" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/run_status.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Viewing all the different run statuses, filtering the runs by status, and refreshing to see latest updates. </figcaption>
						</figure>	
						<div class="callout-block callout-block-info">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-info-circle"></i>
		                            </span><!--//icon-holder-->
	                                Note
	                            </h4>
								<p>Please refresh the project runs page to see the latest updates to your runs!</p>
							</div><!--//content-->
						</div><!--//callout-block-->
					</section><!--//section-->
			    </article><!--//docs-article-->
			    
			    <article class="docs-article" id="section-4">
				    <header class="docs-header">
					    <h1 class="docs-heading">Visualizing results</h1>
					    <section class="docs-intro">
							<p>Once your run is submitted and completed, results are available to visualize and explore. There a variety of interactive visualizations within CReSCENT. Quality control plots include violin plots of QC metrics before and after filtering as well as UMAP plots visualizing the different QC metrics with a gradient. Gene expression can be explored through t-SNE, UMAP, and violin plots. Along with the clustering results from the pipeline, cell metadata previously uploaded is incorporated to colour cells. </p>
							<figure class="figure docs-figure py-3">
								<a href="assets/images/view_run.gif" data-title="Viewing a completed run" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/view_run.gif" alt="" style="width: 800px;"></a>
								<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Viewing a completed run in CReSCENT.</figcaption>
							</figure>					
						</section><!--//docs-intro-->
				    </header>
				     <section class="docs-section" id="item-4-1">
						<h2 class="section-heading">Quality control</h2>
						<p>Quality control in the visualizations page consists of violin plots and UMAPs.</p>
						<h3>Quality control violin plots</h3>
						<p>The QC violin plots are interactive and display a variety of metrics. The four QC metrics (number of genes, number of reads, mitochondrial genes fraction, and ribosomal protein genes fraction) are displayed with distributions before and after QC filtering in the pipeline. </p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/qc_before_after.gif" data-title="Before/after filtering violin plots" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/qc_before_after.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Before/after filtering violin plots and how to switch between viewing barcodes and distribution statistics (mean, median, quartiles, etc.).</figcaption>
						</figure>	
						<h3>Quality control UMAP plots</h3>
						<p>The QC UMAP plots show the intensity of the four QC metrics with a gradient. </p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/qc_umap.gif" data-title="QC UMAP Plots" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/qc_umap.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>QC UMAP plots showing distribution of the QC metrics across the cells.</figcaption>
						</figure>	
					</section><!--//section-->
					
					<section class="docs-section" id="item-4-2">
						<h2 class="section-heading">Gene expression</h2>
						<p>Gene expression can be explored in the visualizations page through t-SNE, UMAP, and violin plots.</p>
						<h3>t-SNE and UMAP</h3>
						<p>Both dimension reduction plots initially show clusters determined by the scRNA-seq pipeline. Cells can also be coloured by the previously uploaded metadata. Selecting a top expressed gene or searching a gene of interest will provide an opacity to the cells. The higher the normalized expression per cell, the darker the opacity. This gene exploration will provide insight into the expression of genes driving a cluster as well as potentially provide enough context to label the cluster.</p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/expression_umap.gif" data-title="QC UMAP Plots" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/expression_umap.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>QC UMAP plots showing distribution of the QC metrics across the cells.</figcaption>
						</figure>	
						<h3>Violin plots</h3>
						<p>The same gene expression exploration (cell metadata colouring, selecting a top expressed gene, and searching a gene of interest) as above can be done through violin plots. </p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/expression_violin.gif" data-title="QC UMAP Plots" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/expression_violin.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>QC UMAP plots showing distribution of the QC metrics across the cells.</figcaption>
						</figure>	
						<div class="callout-block callout-block-info">
                            <div class="content">
                                <h4 class="callout-title">
	                                <span class="callout-icon-holder mr-1">
		                                <i class="fas fa-info-circle"></i>
		                            </span><!--//icon-holder-->
	                                Note
	                            </h4>
								<p>You can quickly toggle through t-SNE/UMAP to violin plots to get a better understanding of the expression of a gene!</p>
							</div><!--//content-->
						</div><!--//callout-block-->
					</section><!--//section-->
			    </article><!--//docs-article-->
			    
			    <article class="docs-article" id="section-5">
				    <header class="docs-header">
					    <h1 class="docs-heading">Downloading results</h1>
					    <section class="docs-intro">
							<p>Downloading results in CReSCENT is easy! After selecting a completed run, clicking the "DOWNLOAD RESULTS" button will give you a zipped folder of your run's results.</p>
							<figure class="figure docs-figure py-3">
								<a href="assets/images/download_results.gif" data-title="Downloading results of a completed run" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/download_results.gif" alt="" style="width: 800px;"></a>
								<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Downloading results of a completed run in CReSCENT.</figcaption>
							</figure>	
							<div class="callout-block callout-block-info">
								<div class="content">
									<h4 class="callout-title">
										<span class="callout-icon-holder mr-1">
											<i class="fas fa-info-circle"></i>
										</span><!--//icon-holder-->
										Note
									</h4>
									<p>Downloaded results are not interactive CReSCENT plots, these are static plots generated from the pipeline.</p>
								</div><!--//content-->
							</div><!--//callout-block-->
						</section><!--//docs-intro-->
				    </header>
				     <section class="docs-section" id="item-5-1">
						<h2 class="section-heading">Failed run logs</h2>
						<p>Downloading failed run logs can help you pinpoint where and why your pipeline has failed. After selecting a failed run, clicking the "DOWNLOAD RESULTS" button will give you a zipped folder of your run's failed run logs. In addition, the pipeline can still be reviewed to see if there was any errors in parameter configuration. </p>
						<figure class="figure docs-figure py-3">
							<a href="assets/images/download_logs.gif" data-title="Downloading logs of a failed run" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/download_logs.gif" alt="" style="width: 800px;"></a>
							<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Downloading logs of a failed run and reviewing submitted parameters in CReSCENT.</figcaption>
						</figure>
					</section><!--//section-->
			    </article><!--//docs-article-->
			    
		        <article class="docs-article" id="section-6">
				    <header class="docs-header">
					    <h1 class="docs-heading">Sharing a project</h1>
					    <section class="docs-intro">
							<p>CReSCENT projects can be shared with collaborators. The shared project and its runs will appear in the collaborator's projects page. Use the "Share Project" button in a project's runs page.</p>
							<figure class="figure docs-figure py-3">
								<a href="assets/images/share_project.gif" data-title="Sharing a project" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/share_project.gif" alt="" style="width: 800px;"></a>
								<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Sharing a project with collaborators in CReSCENT.</figcaption>
							</figure>
							<div class="callout-block callout-block-info">
								<div class="content">
									<h4 class="callout-title">
										<span class="callout-icon-holder mr-1">
											<i class="fas fa-info-circle"></i>
										</span><!--//icon-holder-->
										Note
									</h4>
									<p>Projects can only be shared with users within CReSCENT! Sharing projects needs the collaborator's email registered in CReSCENT. To register, click the user icon in the top right of the portal! </p>
								</div><!--//content-->
							</div><!--//callout-block-->
						</section><!--//docs-intro-->
				    </header>
			    </article><!--//docs-article-->
			    
			    
			    <article class="docs-article" id="section-7">
				    <header class="docs-header">
					    <h1 class="docs-heading">Deleting projects and runs</h1>
					    <section class="docs-intro">
							<p>Deleting projects and runs is easy! Use the "Delete Project" button in a project's runs page to delete a project and all its runs. Use the trash button in the run card to delete an individual run.</p>
							<figure class="figure docs-figure py-3">
								<a href="assets/images/delete_project_run.gif" data-title="Deleting projects and runs" data-toggle="lightbox"><img class="figure-img img-fluid shadow rounded" src="assets/images/delete_project_run.gif" alt="" style="width: 800px;"></a>
								<figcaption class="figure-caption mt-3"><i class="fas fa-info-circle mr-2"></i>Deleting individual runs within a project as well as deleting a project and all its runs.</figcaption>
							</figure>
						</section><!--//docs-intro-->
				    </header>
			    </article><!--//docs-article-->
				<article class="docs-article" id="section-8">
				    <header class="docs-header">
					    <h1 class="docs-heading">Resources</h1>
					    <section class="docs-intro">
							<p>CReSCENT resources will be updated here.</p>
							<div class="col-md-6 col-12">
                                <ul class="list list-unstyled pl-0">
                                    <li><a href="https://crescent.cloud" target="_blank" class="btn btn-primary"><i class="fas fa-play-circle mr-2"></i> Go to crescent.cloud!</a></li>
									<li><a href="https://github.com/pughlab/crescent-frontend" target="_blank" class="btn btn-info"><i class="fab fa-github fa-fw"></i> View Web App Code</a></li>
									<li><a href="https://github.com/pughlab/crescent" target="_blank" class="btn btn-info"><i class="fas fa-code"></i> View Pipeline Scripts</a></li>
                                    <li><a href="#" class="btn btn-secondary"><i class="fas fa-book mr-2"></i> View NAR Paper</a></li>
                                    <li><a href="https://youtu.be/7a23FASMn6I" target="_blank" class="btn btn-danger"><i class="fab fa-youtube"></i> Youtube Tutorial</a></li>
									<li><a href="https://github.com/pughlab/crescent-frontend/issues" target="_blank" class="btn btn-warning"><i class="fas fa-bug mr-2"></i> Report Bugs</a></li>

								</ul>
                            </div>

						</section><!--//docs-intro-->
					</article>
				
			    <article class="docs-article" id="section-9">
				    <header class="docs-header">
					    <h1 class="docs-heading">FAQs</h1>
					    <section class="docs-intro">
							<h5 class="pt-3"><i class="fas fa-question-circle mr-1"></i>Does CReSCENT only support single dataset?</h5>
							<p>Currently, CReSCENT only has single dataset pipelines. We are currently working on an integrate datasets pipeline that will allow for multiple datasets and integration (i.e. clustering and differential expression) using the main datasets as anchors.</p>	
						</section><!--//docs-intro-->
				    </header>
			    </article><!--//docs-article-->
				
			    <footer class="footer">
				    <div class="container text-center py-5">
								<!--/* This template is released under the Creative Commons Attribution 3.0 License. Please keep the attribution link below when using for your own project. Thank you for your support. :) If you'd like to use the template without the attribution, you can buy the commercial license via our website: themes.3rdwavemedia.com */-->
								<small class="copyright">Designed with <i class="fas fa-heart" style="color: #fb866a;"></i> by <a class="theme-link" href="http://themes.3rdwavemedia.com" target="_blank">Xiaoying Riley</a> for developers</small>
								<br>
								<small class="copyright">Template Copyright &copy; <a href="https://themes.3rdwavemedia.com/" target="_blank">3rd Wave Media</a></small>
				        <ul class="social-list list-unstyled pt-4 mb-0">
						    <li class="list-inline-item"><a href="https://github.com/pughlab/crescent-frontend" target="_blank"><i class="fab fa-github fa-fw"></i></a></li> 
				            <li class="list-inline-item"><a href="https://youtu.be/7a23FASMn6I" target="_blank"><i class="fab fa-youtube fa-fw"></i></a></li>
				        </ul><!--//social-list-->
				    </div>
			    </footer>
		    </div> 
	    </div>
    </div><!--//docs-wrapper-->
    
   
       
    <!-- Javascript -->          
    <script src="assets/plugins/jquery-3.4.1.min.js"></script>
    <script src="assets/plugins/popper.min.js"></script>
    <script src="assets/plugins/bootstrap/js/bootstrap.min.js"></script>  
    
    
    <!-- Page Specific JS -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.8/highlight.min.js"></script>
    <script src="assets/js/highlight-custom.js"></script> 
    <script src="assets/plugins/jquery.scrollTo.min.js"></script>
    <script src="assets/plugins/lightbox/dist/ekko-lightbox.min.js"></script> 
    <script src="assets/js/docs.js"></script> 

</body>
</html> 

